envoy/datadog_checks/envoy/parser.py

```diff
@@ -1,37 +1,71 @@
-from .errors import UnknownMetric
+from six.moves import range, zip
+
+from .errors import UnknownMetric, UnknownTags
 from .metrics import METRIC_PREFIX, METRIC_TREE, METRICS
 
 
-def parse_metric(metric):
+def parse_metric(metric, metric_mapping=METRIC_TREE):
     """Takes a metric formatted by Envoy and splits it into a unique
     metric name. Returns the unique metric name, a list of tags, and
     the name of the submission method.
 
     Example:
         'listener.0.0.0.0_80.downstream_cx_total' ->
-        ('listener.downstream_cx_total', ['address:0.0.0.0:80'], 'count')
+        ('listener.downstream_cx_total', ['address:0.0.0.0_80'], 'count')
     """
     metric_parts = []
+    tag_names = []
     tag_values = []
-    mapping = METRIC_TREE
+    tag_builder = []
+    unknown_tags = []
+    num_tags = 0
+    minimum_tag_length = 0
 
     for metric_part in metric.split('.'):
-        if metric_part in mapping:
+        if metric_part in metric_mapping and num_tags >= minimum_tag_length:
+            # Rebuild any built up tags whenever we encounter a known metric part.
+            if tag_builder:
+                for tags in metric_mapping['|_tags_|']:
+                    if num_tags >= len(tags):
+                        break
+
+                constructed_tags = construct_tags(tag_builder, len(tags))
+
+                if tags:
+                    tag_names.extend(tags)
+                    tag_values.extend(constructed_tags)
+                else:
+                    unknown_tags.extend(constructed_tags)
```

Comment:
Why do you want to count unknown tags?

Reply:
For better debugging, note the comment above.